src/java.base/share/classes/java/util/concurrent/ArrayBlockingQueue.java

```diff
@@ -1478,15 +1483,15 @@ public boolean removeIf(Predicate<? super E> filter) {
     /**
      * @throws NullPointerException {@inheritDoc}
      */
-    public boolean removeAll(Collection<? extends @NonNull Object> c) {
+    public boolean removeAll(Collection<?> c) {
```

Comment:
This takes us back to https://github.com/typetools/checker-framework/issues/3197

`removeAll` and `retainAll` officially throw only _both_:
- the argument throws on null queries
- the receiver contains nulls

In this case, the latter condition can't hold. So any call is safe.

[Below](https://github.com/typetools/jdk/pull/71/files#diff-46f7504a7f4f0da0082bf312d4cf1878R663) is a different case: That collection can contain nulls, so the only provably safe argument is another collection that can contain nulls.